tests/rptest/tests/random_node_operations_test.py

```diff
@@ -490,6 +493,37 @@ def enable_write_caching_testing():
         if enable_fast_partition_movement():
             fast_producer_consumer.verify()
 
+        if mixed_versions:
```

Comment:
> a result of membership changes during the test, it was perfectly possible
for the cluster to upgrade its cluster version while executing the test.
This might lead to a test failure as the node with previous version
wasn't able to rejoin the cluster.

just to be sure I'm understanding the issue right, the problem here is that if the random op executor somehow manages to decommission all the prior version nodes, we end up bumping the cluster version and that prevents recommissioning those nodes back?


> Refactored the test to work with mixed versions in a way that the
majority of nodes is not upgraded throughout the test to prevent cluster
version from being updated

I didn't fully follow how this is possible.. I'm guessing it has something to do choosing   `math.ceil(node_count / 2.0)`, op executor doesn't decommission more than half nodes at a time?


Also whats the use of upgrading the cluster in the end? (after the test finished).

Reply:
the idea of this test is to verify if we can execute all operations during the upgrade i.e. when there are brokers with both old and new versions in the same cluster, hence the operations are first executed and then the cluster is fully upgraded. 
It was exactly as you wrote in the first part of the comment i.e. all nodes with old version were decommissioned and the cluster feature manager came to the conclusion that it is possible for the cluster version to move forward as all the members are fully up to date.